service/src/test/java/uk/gov/hmcts/reform/fpl/utils/JudgeAndLegalAdvisorHelperTest.java

```diff
@@ -97,5 +101,85 @@ private void shouldExtractOtherTitleDescriptionWhenOtherTitleSelected() {
 
         assertThat(magistrateName).isEqualTo("His Excellency John Doe");
     }
+
+    @Test
+    void shouldReturnAllocatedJudgeWhenUseAllocatedJudgeSelected() {
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = JudgeAndLegalAdvisor.builder()
+            .judgeTitle(MAGISTRATES)
+            .judgeLastName("Stevenson")
+            .legalAdvisorName("John Papa")
+            .useAllocatedJudge("Yes")
+            .allocatedJudgeLabel("Case assigned to: His Honour Judge Dread")
+            .build();
+
+        Judge allocatedJudge = buildJudge();
+
+        JudgeAndLegalAdvisor expectedJudge = getSelectedJudge(judgeAndLegalAdvisor, allocatedJudge);
+
+        assertThat(expectedJudge.getJudgeTitle()).isEqualTo(HIS_HONOUR_JUDGE);
+        assertThat(expectedJudge.getJudgeLastName()).isEqualTo("Dread");
+        assertThat(expectedJudge.getLegalAdvisorName()).isEqualTo("John Papa");
+    }
+
+    @Test
+    void shouldReturnInputtedJudgeWhenUseAllocatedJudgeNotSelected() {
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = JudgeAndLegalAdvisor.builder()
+            .judgeTitle(MAGISTRATES)
+            .judgeLastName("Stevenson")
+            .legalAdvisorName("John Papa")
+            .useAllocatedJudge("No")
+            .allocatedJudgeLabel("Case assigned to: His Honour Judge Dread")
+            .build();
+
+        Judge allocatedJudge = buildJudge();
+
+        JudgeAndLegalAdvisor expectedJudge = getSelectedJudge(judgeAndLegalAdvisor, allocatedJudge);
+
+        assertThat(expectedJudge.getJudgeTitle()).isEqualTo(MAGISTRATES);
+        assertThat(expectedJudge.getJudgeLastName()).isEqualTo("Stevenson");
+        assertThat(expectedJudge.getLegalAdvisorName()).isEqualTo("John Papa");
+    }
+
+    @Test
+    void shouldBuildAllocatedJudgeLabelWhenAllocatedJudgeIncludesTitleAndLastName() {
+        Judge allocatedJudge = buildJudge();
+
+        String label = buildAllocatedJudgeLabel(allocatedJudge);
+
+        assertThat(label).isEqualTo("Case assigned to: His Honour Judge Dread");
+    }
+
+    @Test
+    void shouldBuildAllocatedJudgeLabelWhenAllocatedJudgeIncludesOtherTitleAndFullName() {
+        Judge allocatedJudge = Judge.builder()
+            .judgeTitle(OTHER)
+            .otherTitle("Mr")
+            .judgeLastName("Watson")
+            .build();
```

Comment:
could be private method buildAllocatedJudge() 

- again not fussed if you don't wish to refactor